lib/internal/modules/esm/resolve.js

```diff
@@ -953,7 +943,13 @@ function moduleResolve(specifier, base, conditions, preserveSymlinks) {
     try {
       resolved = new URL(specifier);
     } catch {
-      ({ resolved, format } = packageResolve(specifier, base, conditions));
+      ({ resolved, format } =
+        amendFormatToUrl(
```

Comment:
Is there a reason we are only amending the format for package resolutions here and not all resolutions (that is we are doing it for `import 'pkg'` but not `import './mod.js'`...)

Reply:
that is just because when I tried this it broke one of the existing tests:
```js
+ actual - expected

+ 'ERR_MODULE_NOT_FOUND'
- 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
       ^
    at [...]node/test/es-module/test-esm-dynamic-import.js:12:12
    at [...]node/test/common/index.js:401:15
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: 'ERR_MODULE_NOT_FOUND',
  expected: 'ERR_UNSUPPORTED_ESM_URL_SCHEME',
  operator: 'strictEqual'
```

It would be indeed better to have the `format` on both resolution paths. 

I added a commit that solves this and makes some exception reordering work to satisfy existing tests. The new test (`test/es-module/test-esm-resolve-type.js`) is updated to expect the resolved `format`.

making it an all resolution paths from `moduleResolve` would break more tests and if needed this should imo be done with an own PR.